### PR TITLE
1259/session store android

### PR DESCRIPTION
**Pull request by Decoder07**
# Description

Added new session store API's

Closes 

- #1259 
- #1252 
---

## Additional context

Need to test tile spotlight and pin message functionality in the application across platforms

### Pre-launch Checklist

- [ ] The [Documentation] is updated accordingly, or this PR doesn't require it.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I added new tests to check the change I am making, or this PR is test-exempt.
- [X] All existing and new tests are passing.

[Documentation]: https://www.100ms.live/docs
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement

### CODE REVIEW

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 1]
```diff
@@ -1182,6 +1215,34 @@ class HmssdkFlutterPlugin :
...
+            }
+        }
+    }
+    private fun addKeyChangeListener(call: MethodCall,result: Result){
+
+        val keys = call.argument<List<String>>("keys") ?: run {
+            HMSErrorLogger.returnArgumentsError("keys parameter is null")
+        }
+
+        keys.let { keys as List<String>
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:1237]: The optional  ` ? ` check for keys is not added here `keys.let`. 
But it is added in the `HMSSessionStoreAction.kt` class below in the `setSessionMetadataForKey` function.
Which one is the intended way?

File: android/src/main/kotlin/live/hms/hmssdk_flutter/methods/HMSSessionStoreAction.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        private fun setSessionMetadataForKey(call: MethodCall,result: Result,hmsSessionStore: HmsSessionStore?){
+            val key = call.argument<String?>("key") ?: run {
+                HMSErrorLogger.returnArgumentsError("key is null")
+            }
+
+            val data = call.argument<String?>("data")
+
+            key?.let { key as String
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/methods/HMSSessionStoreAction.kt:53]: The optional  ` ? ` check for key is added here `key?.let`. 
But it is not added in the `HmssdkFlutterPlugin.kt` class above.
Which one is the intended way?

**Review by ygit — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 3]
```diff
@@ -1182,6 +1215,34 @@ class HmssdkFlutterPlugin :
         }
     }
 
+    private val keyChangeListener = object : HMSKeyChangeListener {
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:1218]: It's possible that apps can attach multiple `HMSKeyChangeListener` for different or even same **keys**. 
With this implementation, it cannot be supported.

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 4]
```diff
@@ -886,6 +908,17 @@ class HmssdkFlutterPlugin :
...
             }
         }
+
+        override fun onSessionStoreAvailable(sessionStore: HmsSessionStore) {
+            val args = HashMap<String, Any?>()
+            args["event_name"] = "on_session_store_available"
+            args["data"] = null
+            hmsSessionStore = sessionStore
+            CoroutineScope(Dispatchers.Main).launch {
+                eventSink?.success(args)
```
[the remark this answers is not in the document]
Comment (Decoder07) [reply android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:918]: This is an `HMSUpdateListener` method call only once to set the `HMSSessionStore` object.

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt (hunk 1 above)
Comment (Decoder07) [reply android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt]: made changes here.

**Review by ygit — APPROVED** [review]
LGTM

**Pull request merged**